dev/Scroller/ScrollerSnapPoint.h

```diff
@@ -41,26 +41,36 @@ class SnapPointBase :
     bool operator< (SnapPointBase* snapPoint);
     bool operator== (SnapPointBase* snapPoint);
 
-    virtual winrt::ExpressionAnimation CreateRestingPointExpression(winrt::Compositor compositor, winrt::hstring target, winrt::hstring scale) = 0;
-    virtual winrt::ExpressionAnimation CreateConditionalExpression(winrt::Compositor compositor, winrt::hstring target, winrt::hstring scale) = 0;
+    virtual winrt::ExpressionAnimation CreateRestingPointExpression(
+        winrt::Compositor const& compositor,
+        winrt::hstring const& target,
+        winrt::hstring const& scale) = 0;
+    virtual winrt::ExpressionAnimation CreateConditionalExpression(
+        winrt::Compositor const& compositor,
+        winrt::hstring const& target,
+        winrt::hstring const& scale) = 0;
     virtual ScrollerSnapPointSortPredicate SortPredicate() = 0;
     virtual void DetermineActualApplicableZone(SnapPointBase* previousSnapPoint, SnapPointBase* nextSnapPoint) = 0;
-    virtual double Influence(double edgeOfMidpoint) = 0;
+    virtual double Influence(double edgeOfMidpoint) const = 0;
     virtual void Combine(winrt::SnapPointBase const& snapPoint) = 0;
-    virtual double Evaluate(double value) = 0;
-    int CombinationCount();
+    virtual double Evaluate(double value) const = 0;
+
+    // Returns True when this snap point is sensitive to the viewport size and is interested in future updates.
+    virtual bool OnViewportChanged(double newViewport) = 0;
```

Comment:
>OnViewportChanged [](start = 17, length = 17)

OnViewportChanged sounds like an event handler, I'd suggest something like, respondsToViewportChanged

Reply:
I renamed it to OnUpdateViewport, similar to OnCreateAutomationPeer.